app/scicrunch_process_results.py

```diff
@@ -70,6 +71,20 @@ def _remove_unused_files_information(obj_list):
 def process_results(results):
     return jsonify({'numberOfHits': results['hits']['total'], 'results': _prepare_results(results)})
 
+# process the search result to get the first scaffold of the first dataset
+def process_get_first_scaffold_info(results):
+    results = _prepare_results(results)
+    # iterate through to get the first scaffold
+    for result in results:
+        if 'abi-scaffold-metadata-file' in result and len(result['abi-scaffold-metadata-file']) > 0:
+            path = result['abi-scaffold-metadata-file'][0].get('dataset', {}).get('path')
+            id = result.get('dataset_identifier', 0)
+            version = result.get('dataset_version', 0)
+            s3uri = result.get('s3uri', Config.DEFAULT_S3_BUCKET_NAME)
+            return jsonify({'path':path, 'id': id, 'version': version, 's3uri': s3uri})
```

Comment:
If this fails you will get data back, but that data will be invalid. Would it be better to wrap this in a a try except block and catch KeyError? Return None if KeyError otherwise return a valid object?

Reply:
I was thinking about that but decide to not do that and let the front end handle that instead. I am keeping it the same as to other SciCrunch APIs as they do not explicitly check for validity as well.

Reply:
Updated.